sssom/util.py

```diff
@@ -965,3 +1000,82 @@ def is_multivalued_slot(slot: str) -> bool:
     # return view.get_slot(slot).multivalued
 
     return slot in multivalued_slots
+
+
+def reconcile_prefix_and_data(
+    msdf: MappingSetDataFrame, prefix_reconciliation: dict
+) -> MappingSetDataFrame:
+    """Reconciles prefix_map and translates CURIE switch in dataframe.
+
+    :param msdf: Mapping Set DataFrame.
+    :param prefix_reconciliation: Prefix reconcilation dictionary from a YAML file
+    :return: Mapping Set DataFrame with reconciled prefix_map and data.
+    """
+    # Discussion about this found here:
+    # https://github.com/mapping-commons/sssom-py/issues/216#issue-1171701052
+
+    prefix_map = msdf.prefix_map
+    df: pd.DataFrame = msdf.df
+    data_switch_dict = dict()
+
+    prefix_synonyms = prefix_reconciliation["prefix_synonyms"]
+    prefix_expansion = prefix_reconciliation["prefix_expansion_reconciliation"]
+
+    # The prefix exists but the expansion needs to be updated.
+    expansion_replace = {
+        k: v
+        for k, v in prefix_expansion.items()
+        if k in prefix_map.keys() and v != prefix_map[k]
+    }
+
+    # Updates expansions in prefix_map
+    prefix_map.update(expansion_replace)
+
+    # Prefixes that need to be replaced
+    # IF condition:
+    #   1. Key OR Value in prefix_synonyms are keys in prefix_map
+    #       e.g.: ICD10: ICD10CM - either should be present within
+    #           the prefix_map.
+    #   AND
+    #   2. Value in prefix_synonyms is NOT a value in expansion_replace.
+    #      In other words, the existing expansion do not match the YAML.
+
+    prefix_replace = [
+        k
+        for k, v in prefix_synonyms.items()
+        if (k in prefix_map.keys() or v in prefix_map.keys())
+        and v not in expansion_replace.keys()
+    ]
+
+    if len(prefix_replace) > 0:
+        for pr in prefix_replace:
+            correct_prefix = prefix_synonyms[pr]
+            correct_expansion = prefix_expansion[correct_prefix]
+            prefix_map[correct_prefix] = correct_expansion
+            logging.info(f"Adding prefix_map {correct_prefix}: {correct_expansion}")
+            if pr in prefix_map.keys():
+                prefix_map.pop(pr, None)
+                data_switch_dict[pr] = correct_prefix
+
+                logging.warning(f"Replacing prefix {pr} with {correct_prefix}")
+
+    # Data editing
+    if len(data_switch_dict) > 0:
+        # Read schema file
+        with open(SCHEMA_YAML) as file:
```

Comment:
Isnt it better to use schemaview? I feel this here would break every time the yaml file changes a bit..

Reply:
I can change it to use `SchemaView` but am not sure how this will break. I wasn't aware of `SchemaView` at the time of writing this.

Reply:
Done!